### PR TITLE
Copy the RunScript command into Sashimi

### PR DESCRIPTION
**Pull request by flin-8**
Calamari PR: https://github.com/OctopusDeploy/Calamari/pull/649

This is a copy of RunScript using the new PipelineCommand structure. This allows the different Azure flavours to share this code.

We still need to decide who is responsible for registering or discovering the dependencies

### CODE REVIEW

File: source/Calamari.Scripting/Calamari.Scripting.csproj [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <Nullable>enable</Nullable>
+        <OutputType>Exe</OutputType>
+        <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+        <TargetFrameworks>net452;netcoreapp3.1</TargetFrameworks>
```
Comment (johnsimons) [inline source/Calamari.Scripting/Calamari.Scripting.csproj:7]: No net40?

File: source/Calamari.Scripting/Program.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using Calamari.Common;
+using Calamari.Common.Plumbing.Logging;
+
+namespace Calamari.Scripting
+{
+    public class Program : CalamariFlavourProgram
```
Comment (johnsimons) [inline source/Calamari.Scripting/Program.cs:7]: Should this be deriving from `CalamariFlavourProgramAsync`?

File: source/Calamari.Scripting/Calamari.Scripting.csproj (hunk 1 above)
Comment (flin-8) [reply source/Calamari.Scripting/Calamari.Scripting.csproj:7]: I think it has to be net452 for now, Server doesn't support multiple netfx versions atm.... @slewis74 right?

File: source/Calamari.Scripting/Program.cs (hunk 2 above)
Comment (flin-8) [reply source/Calamari.Scripting/Program.cs:7]: Yup 😁

File: source/Calamari.Scripting/Calamari.Scripting.csproj (hunk 1 above)
Comment (slewis74) [reply source/Calamari.Scripting/Calamari.Scripting.csproj:7]: If this is a flavour then it should be net40. Server can only support one or other of net40 and net452 for a flavour. For a library we'd add both.

File: source/Calamari.Scripting/Program.cs (hunk 2 above)
Comment (johnsimons) [reply source/Calamari.Scripting/Program.cs:7]: well if we are making this a lib only for now, I'll remove the program.cs

File: source/Sashimi.AzureWebApp/AzureWebAppScriptActionOverride.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
```
Comment (flin-8) [inline source/Sashimi.AzureWebApp/AzureWebAppScriptActionOverride.cs:1]: This has moved

**Comment by flin-8** [comment]
Looks good, We will need to re-add the script overrides to CloudService and WebApp.

I can't approve my own PR

File: source/Sashimi.AzureWebApp/AzureWebAppScriptActionOverride.cs (hunk 3 above)
Comment (johnsimons) [reply source/Sashimi.AzureWebApp/AzureWebAppScriptActionOverride.cs:1]: good point, i'll get rid of it

**Pull request merged**